### PR TITLE
Minor using decl fixes

### PR DESCRIPTION
**Pull request by chsienki**
Some minor cleanup for using declarations

Fixes https://github.com/dotnet/roslyn/issues/28892
Fixes https://github.com/dotnet/roslyn/issues/32318

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Parser/LanguageParser.cs [hunk 1]
```diff
@@ -6522,11 +6522,11 @@ bool isPossibleAwaitUsing()
                     this.PeekToken(1).Kind == SyntaxKind.UsingKeyword;
             }
 
-            SyntaxToken parseAwaitKeywordForAsyncStreams()
+            SyntaxToken parseAwaitKeyword(MessageID? feature = null)
             {
                 Debug.Assert(this.CurrentToken.ContextualKind == SyntaxKind.AwaitKeyword);
                 SyntaxToken awaitToken = this.EatContextualToken(SyntaxKind.AwaitKeyword);
-                return CheckFeatureAvailability(awaitToken, MessageID.IDS_FeatureAsyncStreams);
+                return (feature.HasValue) ? CheckFeatureAvailability(awaitToken, feature.Value) : awaitToken;
```
Comment (cston) [inline src/Compilers/CSharp/Portable/Parser/LanguageParser.cs]: >(feature.HasValue) [](start = 23, length = 18)

Parentheses are not needed.

Comment (cston) [inline src/Compilers/CSharp/Portable/Parser/LanguageParser.cs]: >Value [](start = 89, length = 5)

Can use `GetValueOrDefault()` since we've checked `HasValue`.

File: src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncIteratorTests.cs [hunk 2]
```diff
@@ -861,15 +861,8 @@ public interface IAsyncEnumerator<out T> : System.IAsyncDisposable
...
-namespace System
-{
-    public interface IAsyncDisposable
-    {
-        System.Threading.Tasks.ValueTask DisposeAsync();
-    }
-}
 ";
-            var comp = CreateCompilationWithTasksExtensions(new[] { source });
+            var comp = CreateCompilationWithTasksExtensions(new[] { source, CSharpTestBase.IAsyncDisposableDefinition });
```
Comment (cston) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncIteratorTests.cs]: >CSharpTestBase [](start = 76, length = 14)

Is type qualifier needed?

File: src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAwaitUsingTests.cs [hunk 3]
```diff
@@ -39,11 +29,11 @@ public System.Threading.Tasks.ValueTask DisposeAsync()
     }
 }
 ";
-            var comp = CreateCompilationWithTasksExtensions(source + s_interfaces, parseOptions: TestOptions.Regular7_3);
+            var comp = CreateCompilationWithTasksExtensions(source + CSharpTestBase.IAsyncDisposableDefinition, parseOptions: TestOptions.Regular7_3);
```
Comment (cston) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAwaitUsingTests.cs]: >source + CSharpTestBase.IAsyncDisposableDefinition [](start = 60, length = 50)

Since you're changing all these, consider using `new[] { source, IAsyncDisposableDefinition }` instead if there's no reason to concatenate the two source files.

File: src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs [hunk 4]
```diff
@@ -182,12 +182,11 @@ public sealed class NotNullIfNotNullAttribute : Attribute
 ";
 
         protected const string IAsyncDisposableDefinition = @"
-using System.Threading.Tasks;
 namespace System
 {
     public interface IAsyncDisposable
     {
-        ValueTask DisposeAsync();
+        System.Threading.Tasks.ValueTask DisposeAsync();
```
Comment (cston) [inline src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs]: >System.Threading.Tasks. [](start = 8, length = 23)

Can we revert this change if the sources are no longer concatenated?

File: src/Compilers/CSharp/Portable/Parser/LanguageParser.cs [hunk 5]
```diff
@@ -6469,17 +6469,17 @@ private StatementSyntax ParseStatementNoDeclaration(bool allowAnyExpression)
...
                     {
                         if (PeekToken(2).Kind == SyntaxKind.OpenParenToken)
                         {
-                            return this.ParseUsingStatement(parseAwaitKeywordForAsyncStreams());
+                            return this.ParseUsingStatement(parseAwaitKeyword(MessageID.IDS_FeatureAsyncUsing));
                         }
                         else
                         {
-                            return this.ParseLocalDeclarationStatement(parseAwaitKeywordForAsyncStreams());
+                            return this.ParseLocalDeclarationStatement(parseAwaitKeyword());
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Portable/Parser/LanguageParser.cs]: is this missing a possible LangVersion diagnostic for `using var`? Or is it reported elsewhere?

Comment (chsienki) [reply src/Compilers/CSharp/Portable/Parser/LanguageParser.cs]: No, previously we were reporting for both the `await` and the `using`. This change removes the `await` warning in this path so just the using gets the lang ver warning.
